### PR TITLE
acl: add DisplayName field to auth methods

### PR DESCRIPTION
**Pull request by rboyer**
Also add a few missing acl fields in the api.

### CODE REVIEW

File: command/acl/authmethod/update/authmethod_update_test.go [hunk 1]
```diff
@@ -726,3 +742,28 @@ func TestAuthMethodUpdateCommand_k8s_noMerge(t *testing.T) {
...
+
+	method, _, err := client.ACL().AuthMethodRead(
+		methodName,
+		&api.QueryOptions{Token: "root"},
+	)
+	require.NoError(t, err)
+	require.NotNil(t, method)
+
+	// zero these out since we don't really care
+	method.CreateIndex = 0
```
Comment (s-christoff) [inline command/acl/authmethod/update/authmethod_update_test.go:757]: Why are these here?

Comment (rboyer) [reply command/acl/authmethod/update/authmethod_update_test.go:757]: Because it's hard to use `require.Equals` if you don't know what these two field values are set to. For all of these tests I'm just setting these to  zero so I can test the rest of the fields round tripped easily.

Comment (s-christoff) [reply command/acl/authmethod/update/authmethod_update_test.go:757]: o til thank you :)

File: website/pages/api-docs/acl/auth-methods.mdx [hunk 2]
```diff
@@ -40,13 +40,18 @@ The table below shows this endpoint's support for
...
+
 - `Type` `(string: <required>)` - The type of auth method being configured.
   The only allowed value in Consul 1.5.0 is `"kubernetes"`. This field is
   immutable.
 
 - `Description` `(string: "")` - Free form human readable description of the
   auth method.
 
+- `DisplayName` `(string: "")` - An optional name to use instead of the `Name`
+  field when displaying information about this auth method in any kind of user
```
Comment (s-christoff) [inline website/pages/api-docs/acl/auth-methods.mdx:52]: `in any kind of user interface` isn't this primarily used in the UI and not CLI and we should specify that as much? 🤔

Comment (rboyer) [reply website/pages/api-docs/acl/auth-methods.mdx:52]: Well you could roll your own wrapper on top of consul if you didn't want to use our UI.

I could just end the sentence at "...about this auth method." instead to keep it vague.

Comment (s-christoff) [reply website/pages/api-docs/acl/auth-methods.mdx:52]: o ok sgtm
